src/Main.elm

```diff
@@ -46,6 +51,17 @@ init =
         watchDate movie =
             Movie.watchDate movie
                 |> Maybe.withDefault (date 1800 1 1)
+
+        genres =
+            MovieList.movies
+                |> List.map .genres
+                |> List.map (Set.map String.toLower)
+                |> List.foldr Set.union Set.empty
```

Comment:
Like reduce. Here, there are no parens, because `List.foldr` takes 3 arguments: a function to run on each input item, an initial value, and (piped in from previous line, a List)